instrumentation/spring-webmvc/src/main/java/brave/spring/webmvc/TracingHandlerInterceptor.java

```diff
@@ -68,6 +81,19 @@ public void afterCompletion(HttpServletRequest request, HttpServletResponse resp
     Span span = tracer.currentSpan();
     if (span == null) return;
     ((SpanInScope) request.getAttribute(SpanInScope.class.getName())).close();
+    Object template = request.getAttribute(BEST_MATCHING_PATTERN_ATTRIBUTE);
+    if (template != null) {
+      response = new HttpServletResponseWithTemplate(response, template.toString());
+    }
     handler.handleSend(response, ex, span);
   }
+
+  static class HttpServletResponseWithTemplate extends HttpServletResponseWrapper {
```

Comment:
cc @llinder @hyleung When an http library provides a wrapper, we get a *big* benefit, which is the ability to stash things through response-shaped holes. You can look at the note on vert.x for the opposite (not saying vert.x is bad, just that wrappers are safer for us to use)

cc @cescoffier

Reply:
I don't think Ratpack has wrappers out-of-the-box, but iirc we had to _introduce_ wrapper types to work around Ratpack's types (e.g. we don't really get a "Request", but a `RequestSpec` - so we have to wrap it in something that is more request-like).

Reply:
very confident this api is fine for ratpack https://github.com/hyleung/ratpack-zipkin/commit/bb498012771bc11ae32a1be1c6fcf7c6bb2b3ab9